### ISSUE TITLE
关于代码的错误

### ISSUE THREAD

**Issue by twiQ1** [1e12Leon/PDIKD#1]
class FourierTransform(nn.Module):
    def __init__(self):
        super(FourierTransform, self).__init__()

    def forward(self, features):
        return self.fourier_transform(features)

    def fourier_transform(self, features):
        """
        对特征进行傅里叶变换，并返回高频特征部分。
        :param features: 输入特征，shape 可以是 (80, 80, 256), (40, 40, 512) 或 (20, 20, 1024)
        :return: 高频特征部分
        """
        # 对特征进行傅里叶变换
        transformed = torch.fft.fft2(features, dim=(-3, -2))

        # 移动零频分量到中心
        shifted = torch.fft.fftshift(transformed, dim=(-3, -2))

        # 获取特征的形状
        h, w = features.shape[-3], features.shape[-2]

        # 计算高频区域的边界
        high_freq_bound_h = h // 4
        high_freq_bound_w = w // 4

        # 提取高频特征
        high_freq = shifted.clone()
        high_freq[:high_freq_bound_h, :, :] = 0
        high_freq[-high_freq_bound_h:, :, :] = 0
        high_freq[:, :high_freq_bound_w, :] = 0
        high_freq[:, -high_freq_bound_w:, :] = 0

        # 逆傅里叶变换
        high_freq_shifted = torch.fft.ifftshift(high_freq, dim=(-3, -2))
        high_freq_features = torch.fft.ifft2(high_freq_shifted, dim=(-3, -2))

        return high_freq_features.abs()          abs是振幅吧？  和原论文对齐相位的说法不符？
